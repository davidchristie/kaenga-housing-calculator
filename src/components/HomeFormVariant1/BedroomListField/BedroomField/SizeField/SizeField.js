import React, { Component, PropTypes } from 'react'
import {
  ControlLabel,
  FormControl,
  FormGroup
} from 'react-bootstrap'

class SizeField extends Component {
  render () {
    const { input: { onChange, value } } = this.props
    return (
      <FormGroup>
        <ControlLabel>How big is the bedroom?</ControlLabel>
        <FormControl
          componentClass='select'
          onChange={onChange}
          value={value}
        >
          <option value='double'>Double</option>
          <option value='single'>Single</option>
        </FormControl>
      </FormGroup>
    )
  }
}

SizeField.defaultProps = {
  input: {}
}

SizeField.propTypes = {
  input: PropTypes.shape({
    onChange: PropTypes.func,
    value: PropTypes.string
  }).isRequired
}

export default SizeField
