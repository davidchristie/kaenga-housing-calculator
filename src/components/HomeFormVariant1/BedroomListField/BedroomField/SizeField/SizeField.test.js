import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import SizeField from './SizeField'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(shallowRenderer.render(<SizeField />)).toMatchSnapshot()
})
