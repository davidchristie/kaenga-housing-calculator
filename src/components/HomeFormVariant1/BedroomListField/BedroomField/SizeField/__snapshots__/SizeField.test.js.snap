exports[`test matches snapshot 1`] = `
<FormGroup
  bsClass="form-group">
  <ControlLabel
    bsClass="control-label"
    srOnly={false}>
    How big is the bedroom?
  </ControlLabel>
  <FormControl
    bsClass="form-control"
    componentClass="select"
    onChange={undefined}
    value={undefined}>
    <option
      value="double">
      Double
    </option>
    <option
      value="single">
      Single
    </option>
  </FormControl>
</FormGroup>
`;
